State mismatch during component update

### Current Situation

Currently, updating any child component state can result in `AssertionError: state mismatch during component update - key '61e47127-9d3f-4954-8235-4d99122134c1'!=4`

This can be previewed within Conreq's app store branch.

### Proposed Actions

Update `layout.py` to not cause this exception.

### Work Items

- [ ] Develop a fix for state mismatch
- [ ] Release 0.35.5 to resolve this

--- a/src/idom/core/layout.py
+++ b/src/idom/core/layout.py
@@ -216,11 +216,6 @@ class Layout:
             pass
         else:
             key, index = new_state.key, new_state.index
-            if old_state is not None:
-                assert key == old_state.key, (
-                    "state mismatch during component update - "
-                    f"key {key!r}!={old_state.key!r} "
-                )
             parent.children_by_key[key] = new_state
             # need to do insertion in case where old_state is None and we're appending
             parent.model.current["children"][index : index + 1] = [

